src/docfx/build/toc/parser/TableOfContentsParser.cs

```diff
@@ -213,12 +213,6 @@ string GetTopicHref(TableOfContentsInputItem tocInputModel)
                     return (tocHref, default, default);
                 }
 
-                var (hrefPath, fragment, query) = HrefUtility.SplitHref(tocHref);
-                if (!string.IsNullOrEmpty(fragment) || !string.IsNullOrEmpty(query))
-                {
-                    // '#' and '?' is not allowed when referencing toc file
-                    tocHref = hrefPath;
-                }
```

Comment:
why remove this logic?

Reply:
ResolveContent will split href internally, so this is unnecessary

Reply:
what if the href is like 'a/b/?branch=master'

---
In reply to: [197993285](https://github.com/dotnet/docfx/pull/2934#discussion_r197993285) [](ancestors = 197993285)

Reply:
You are right 👍